src/containers/tud/TimeUseDiarySagas.js

```diff
@@ -213,17 +220,67 @@ function* getSubmissionsByDateWatcher() :Saga<*> {
   yield takeEvery(GET_SUBMISSIONS_BY_DATE, getSubmissionsByDateWorker);
 }
 
-function* downloadTudResponsesWorker(action :SequenceAction) :Saga<WorkerResponse> {
+function* downloadSummarizedData(entities, outputFilename) :Saga<WorkerResponse> {
+  let workerResponse = {};
+  try {
+
+    const submissionIds = entities.map((entity) => entity.get(OPENLATTICE_ID_FQN)).flatten();
+    const submissionMetadata = Map(entities.map((entity) => [entity.getIn([OPENLATTICE_ID_FQN, 0]), entity]));
+
+    // entity set ids
+    const surveyEntitySets :Map = yield select(selectEntitySetsByModule(AppModules.QUESTIONNAIRES));
+
+    const submissionESID = surveyEntitySets.get(SUBMISSION);
+    const registeredForESID = surveyEntitySets.get(REGISTERED_FOR);
+    const summarysetESID = surveyEntitySets.get(SUMMARY_SET);
+
+    // filtered search on submissions
+    const filter = {
+      destinationEntitySetIds: [submissionESID],
+      edgeEntitySetIds: [registeredForESID],
+      entityKeyIds: submissionIds.toJS(),
+      sourceEntitySetIds: [summarysetESID]
+    };
+
+    const response = yield call(
+      searchEntityNeighborsWithFilterWorker,
+      searchEntityNeighborsWithFilter({
+        entitySetId: submissionESID,
+        filter,
+      })
+    );
+    if (response.error) throw response.error;
+
+    const csvHeaders = OrderedSet().asMutable();
+
+    // create submissionEKID -> ol.variable -> ol.value map
+    const summaryData = Map().withMutations((mutator) => {
+      fromJS(response.data).forEach((neighbors, submissionId) => {
+        neighbors.forEach((neighbor) => {
+          const neighborDetails = neighbor.get('neighborDetails');
+          const variable = getPropertyValue(neighborDetails, [VARIABLE_FQN, 0]);
+          const value = getPropertyValue(neighborDetails, [VALUES_FQN, 0]);
+          mutator.setIn([submissionId, variable], value);
+          csvHeaders.add(variable);
+        });
+      });
+    });
+
+    exportSummarizedDataToCsvFile(summaryData, submissionMetadata, csvHeaders.asImmutable(), outputFilename);
+
+    workerResponse = { data: null };
```

Comment:
Setting it to `null` seems like an easy way to introduce a bug in the future.

```suggestion
    workerResponse = { data: {} };
```